pkgs/development/compilers/go/1.15.nix

```diff
@@ -11,9 +11,9 @@ let
 
   inherit (lib) optionals optionalString;
 
-  go_bootstrap = callPackage ./bootstrap.nix {
-    inherit Security;
-  };
+  version = "1.15.8";
+
+  go_bootstrap = callPackage ./bootstrap.nix { };
```

Comment:
How do you intend this to work so we can keep 1.13 bootstrap for 1.15 and we add a 1.16 bootstrap for 1.16?

Reply:
We can pass `version` here in inputs and check it in `bootstrap.nix` to select the appropriate bootstrap version.

If go1.16 can bootstrap everything all the same, is there a reason not to use it?

Reply:
> If go1.16 can bootstrap everything

Does using 1.16 to bootstrap 1.15 work on all platforms?

Reply:
Over on the installing from source [instructions](https://golang.org/doc/install/source#bootstrapFromBinaryRelease) they say that we can use any version as bootstrap so I think that we can use any version of Go and we don't have to use one per version. IIRC the bootstrap is used to build the compiler then the compiler is then used to build itself anyway.

Reply:
I don't know for sure (but suspect yes), that's why passing `version` is a backup plan.

Reply:
Another reason to keep separate bootstraps for now: just updating the bootstrap to add `go_1_16` (without changing the default `go` version) will cause a rebuild of all go packages.

Reply:
We could just wait to merge this after we've added `go_1_16` and made it the default?

Reply:
> Another reason to keep separate bootstraps for now: just updating the bootstrap to add `go_1_16` (without changing the default go version) will cause a rebuild of all go packages.

I'm not sure why we need to update bootstrap to add `go_1_16` until go1.16 is the default.

The timeline I have in mind:

1. Merge this, rebuilding all Go packages.
2. Merge #107937, building packages for Go 1.16 (bootstrapped from Go 1.13).
3. Merge a PR that changes both bootstrap and default Go versions to 1.16, rebuilding all packages.

I don't really see any convenient scenarios with fewer than two full rebuilds between current situation and Go 1.16 being the default with this PR merged.

Reply:
If we do 1 and 3 at the same time it's one rebuild.

Reply:
I'm ok with it if you think that saving one rebuild is worth it.

Reply:
Why do we need to move the bootstrap to go 1.16 at all? 🤔

Reply:
To bootstrap on `aarch64-darwin`.

Reply:
We'll definitely need it for `aarch64-darwin`, since no release other than 1.16 can bootstrap it. Having the same version for everything else is probably nice for consistency.

Either way, that doesn't really change the number of rebuilds required with this PR.

Reply:
Ahh I see, yea that makes sense. I'm ok waiting till we get 1.16 merged before merging this. No harm keeping it aside for now...

Reply:
@zowoq should I update this PR to 1.16 then?

Reply:
@bobrik since 1.16 was released then yes, please. I'll merge this as soon as #107937 is updated to 1.16 and merged so hydra rebuilds both together.

Reply:
> I'll merge this as soon as #107937 is updated to 1.16

We still need to change the defaults to 1.16 here and test them first.

Reply:
Updated (including `darwin-arm64`).

Reply:
Please rebase on master and apply the bootstrap changes to `1.16.nix`.

To change the defaults to `go_1_16` you can cherry-pick https://github.com/zowoq/nixpkgs/commit/76e01e2b6d1744661cd0ed28525b5d9bfbac18fe or I can push it to this branch for you.

Reply:
Sorry, wrong commit. Should have been https://github.com/zowoq/nixpkgs/commit/8f3a96cfd043704a6d227b13e867c9528b1e522e.

Reply:
Rebased and cherry-picked.